File: Problemas/749A - Bachgold Problem/749A_Bachgold_Problem.py
# 749A - Bachgold Problem
# https://codeforces.com/problemset/problem/749/A

# Input:
# 1) N = Número a descomponer
N = int(input())

# Calcula la cantidad de números primos a utilizar
numeros_primos = N // 2

# Imprime la cantidad de números primos a utilizar
print(numeros_primos)

# Números primos
string_primos = ""

# Itera mientras N sea mayor o igual a 0
while N > 0:

    # Si N es impar, entonces resta -3 para convertirlo a par
    if N % 2 == 1:
        string_primos += "3 "
        N -= 3

    # Si N es par, entonces resta -2 hasta llegar a 0
    else:
        string_primos += "2 "
        N -= 2

# Imprimer los números primos que suman N
print(string_primos)

File: Problemas/749A - Bachgold Problem/749A_Bachgold_Problem.cpp
// 749A - Goldbach Problem
// https://codeforces.com/problemset/problem/749/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) N = Número a descomponer
    int N;
    cin >> N;

    // Calcula la cantidad de números primos a utilizar
    int numeros_primos = N / 2;

    // Imprime la cantidad de números primos a utilizar
    cout << numeros_primos << endl;

    // Si N es impar, entonces resta -1 a la cantidad de números a utilizar
    // El número primo que se elimina del total es la resta -3 a N
    if(N % 2 == 1) {
        cout << "3 ";
        numeros_primos--;
    }

    // Itera sobre la cantidad de número primos a utilizar
    // Por cada número a utilizar, se resta -2 a N
    for(int numero = 0; numero < numeros_primos; numero++) {
        cout << "2 ";
    }

    return 0;
}